robot_hardware:
  - upper_body
  - lower_body
  - i2c
upper_body:
  type: mh5_hardware/MH5DynamixelInterface
  joints:
    - head_p
    - head_y
    - l_sho_p
    - l_sho_r
    - l_elb_y
    - l_elb_p
    - r_sho_p
    - r_sho_r
    - r_elb_y
    - r_elb_p
  port: /dev/ttySC0
  baudrate: 2000000
  protocol: 2.0
  rs485: true
  rates:
    pvl_reader: 100.0
    pv_writer: 100.0
    t_writer: 2.0
  head_p:
    id: 51
    inverse: true
  head_y:
    id: 52
  l_sho_p:
    id: 34
    inverse: true
  l_sho_r:
    id: 33
    inverse: true
  l_elb_y:
    id: 32
    inverse: true
  l_elb_p:
    id: 31
  r_sho_p:
    id: 44
  r_sho_r:
    id: 43
  r_elb_y:
    id: 42
  r_elb_p:
    id: 41
    inverse: true
lower_body:
  type: mh5_hardware/MH5DynamixelInterface
  port: /dev/ttySC1
  baudrate: 2000000
  protocol: 2.0
  rs485: true
  rates:
    pvl_reader: 100.0
    pv_writer: 100.0
    t_writer: 2.0
  joints:
    - l_hip_r
    - l_hip_p
    - l_kne_p
    - l_kne_y
    - l_ank_p
    - l_ank_r
    - r_hip_r
    - r_hip_p
    - r_kne_p
    - r_kne_y
    - r_ank_p
    - r_ank_r
  l_hip_r:
    id: 16
    inverse: true
  l_hip_p:
    id: 15
    inverse: true
  l_kne_p:
    id: 14
    inverse: true
    offset: 1.570796326794897
  l_kne_y:
    id: 13
    inverse: true
  l_ank_p:
    id: 12
    inverse: true
  l_ank_r:
    id: 11
  r_hip_r:
    id: 26
  r_hip_p:
    id: 25
    inverse: true
  r_kne_p:
    id: 24
    inverse: true
    offset: 1.570796326794897
  r_kne_y:
    id: 23
  r_ank_p:
    id: 22
    inverse: true
  r_ank_r:
    id: 21
    inverse: true
i2c:
  type: mh5_hardware/MH5I2CInterface
  port: /dev/i2c-1
  rates:
    imu: 200.0
    adc: 1.0
  imu:
    id: 106
    lpf: 0.1
    # orientation: [ 0.7071068, 0.0, -0.7071068, 0.0 ]
